src/react/components/StyleEditor/Preview.jsx

```diff
@@ -190,47 +115,126 @@ export class Preview extends React.Component {
+        const widthV = previewViewbox.width - (5 * size); // multiply by negative to shrink viewbox
+        const heightV = previewViewbox.height - (5 * size); // multiply by negative to shrink viewbox
         viewboxString = previewViewbox.minX + ' ' + previewViewbox.minY + ' ' +  widthV + ' ' + heightV;
         return viewboxString;
     }
 
     /**
-     * @method _addBaseSvg
-     *
-     * @returns {Component} - combined svg icon with preview base wrapping around provided icon
-     *
+     * @method _composePointPath
+     * @description Composes point svg path
+     * @returns {String} point svg path
+     */
+    const _composePointPath = () => {
+        const path = _parsePath();        
+
+        previewAttributes.strokeColor = oskariStyle.stroke.color;
+        previewAttributes.fillColor = oskariStyle.fill.color;
+        previewAttributes.fill = oskariStyle.image.fill.color;
+        previewAttributes.strokeWidth = PREVIEW_DEFAULTS.defaultStrokeWidth;
+        previewAttributes.strokeLineCap = PREVIEW_DEFAULTS.stroke.lineCap;
+        previewAttributes.strokeDashArray = _getPreviewLineDash();
+        previewAttributes.strokeLineJoin = PREVIEW_DEFAULTS.stroke.area.lineJoin;
+
+        path.setAttribute('stroke', previewAttributes.strokeColor);
+        path.setAttribute('stroke-width', previewAttributes.strokeWidth);
+        path.setAttribute('stroke-linecap', previewAttributes.strokeLineCap);
+        path.setAttribute('stroke-dasharray', previewAttributes.strokeDashArray);
+        path.setAttribute('stroke-linejoin', previewAttributes.strokeLineJoin);
+        path.setAttribute('fill', previewAttributes.fill);
+        
+        size = oskariStyle.image.size;
+
+        return path.outerHTML;
+    }
+
+    /**
+     * @method _composeLinePath
+     * @description Composes line svg path
+     * @returns {String} line svg path
+     */
+    const _composeLinePath = () => {
+        const path = _parsePath();        
+
+        previewAttributes.strokeColor = oskariStyle.stroke.color;
+        previewAttributes.fillColor = oskariStyle.fill.color;
+        previewAttributes.fill = PREVIEW_DEFAULTS.fill.color;
+
+        previewAttributes.strokeWidth = oskariStyle.stroke.width;
+        previewAttributes.strokeLineCap = oskariStyle.stroke.lineCap;
+        previewAttributes.strokeDashArray = _getPreviewLineDash();
+        previewAttributes.strokeLineJoin = oskariStyle.stroke.area.lineJoin;
+
+        path.setAttribute('stroke', previewAttributes.strokeColor);
+        path.setAttribute('stroke-width', previewAttributes.strokeWidth);
+        path.setAttribute('stroke-linecap', previewAttributes.strokeLineCap);
+        path.setAttribute('stroke-dasharray', previewAttributes.strokeDashArray);
+        path.setAttribute('stroke-linejoin', previewAttributes.strokeLineJoin);
+
+        size = PREVIEW_DEFAULTS.image.size;
+
+        return path.outerHTML;
+    }
```

Comment:
If you add oskariStyle as param to _composeLinePath() it doesn't need to be defined inside the render function. If it's not inside the render function it's easier to debug and test by itself helping with maintenance etc.

Reply:
Moved along all the other helper functions outside of render.